### PR TITLE
Bug 1068887 - Updated visibility settings on recording timer

### PR DESCRIPTION
**Pull request by funktr0n**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[FunkTron FunkTron](https://github.com/FunkTron) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=75adebbdb6d6)

**Comment by try-server-hook** [comment]
[FunkTron FunkTron](https://github.com/FunkTron) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=54ac4c49aefc)

**Comment by try-server-hook** [comment]
[FunkTron FunkTron](https://github.com/FunkTron) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=b53c92b2c86b)

File: apps/camera/style/recording-timer.css [hunk 1]
```diff
@@ -5,6 +5,7 @@
   left: 0;
   z-index: 1;
 
+  visibility: hidden;
```
Comment (yzen) [inline apps/camera/style/recording-timer.css]: So you also need to make sure that visibility works well with opacity transition. See https://github.com/mozilla-b2g/gaia/blob/master/apps/camera/style/hud.css for an example.

File: apps/camera/test/unit/controllers/recording-timer_test.js [hunk 2]
```diff
@@ -10,11 +10,22 @@ suite('controllers/recording-timer', function() {
     ], function(App, RecordingTimerController, RecordingTimerView) {
       self.RecordingTimerController = RecordingTimerController.RecordingTimerController;
       self.RecordingTimerView = RecordingTimerView;
+      self.style = loadCss('/style/recording-timer.css',
```
Comment (yzen) [inline apps/camera/test/unit/controllers/recording-timer_test.js]: Will it work if you just pass done as a second arg?

File: apps/camera/test/unit/controllers/recording-timer_test.js [hunk 3]
```diff
@@ -53,4 +64,33 @@ suite('controllers/recording-timer', function() {
...
+    setup(function() {
+      app = sinon.createStubInstance(this.App);
+      app.view = new this.RecordingTimerView();
+      view = app.view;
+      controller = new this.RecordingTimerController(app);
+    });
+
+    test('Explicitly test for RecordingTimer visibility', function() {
+      // App is in its initialized state
+      assert.equal(window.getComputedStyle(view.el, null).visibility, 'hidden');
```
Comment (yzen) [inline apps/camera/test/unit/controllers/recording-timer_test.js]: I think it's sufficient to just test for visible class being present/not present in these 3 cases.

File: apps/camera/style/recording-timer.css (hunk 1 above)
Comment (FunkTron) [inline apps/camera/style/recording-timer.css]: The example you gave only works one way... Meaning that you'll only be able to notice the element fading out, not fading in.  Is that the desired behavior?

File: apps/camera/test/unit/controllers/recording-timer_test.js (hunk 3 above)
Comment (FunkTron) [inline apps/camera/test/unit/controllers/recording-timer_test.js]: Doesn't it already do that (more or less) here: https://github.com/mozilla-b2g/gaia/blob/master/apps/camera/test/unit/controllers/recording-timer_test.js#L44-L48?

File: apps/camera/style/recording-timer.css (hunk 1 above)
Comment (yzen) [inline apps/camera/style/recording-timer.css]: I think it actually works both ways, notice how transition delays are handled in both cases. Yours is slightly different as your visibility is inversed, so you would need to place/adjust the delays accordingly.

File: apps/camera/test/unit/controllers/recording-timer_test.js (hunk 2 above)
Comment (FunkTron) [inline apps/camera/test/unit/controllers/recording-timer_test.js]: No, it gives me an error: "Done() invoked with non-error [object] object"

Comment (yzen) [inline apps/camera/test/unit/controllers/recording-timer_test.js]: Makes sense, what about done.bind(this)? Just curious..

File: apps/camera/test/unit/controllers/recording-timer_test.js (hunk 3 above)
Comment (yzen) [inline apps/camera/test/unit/controllers/recording-timer_test.js]: Yeah good point, I would still check that the class is there perhaps, the test you have is more like an integration test that we write with marionette..

File: apps/camera/test/unit/controllers/recording-timer_test.js (hunk 2 above)
Comment (FunkTron) [inline apps/camera/test/unit/controllers/recording-timer_test.js]: Nope, same error

File: apps/camera/style/recording-timer.css (hunk 1 above)
Comment (FunkTron) [inline apps/camera/style/recording-timer.css]: Yeah I see now, was reading it the wrong way initially

File: apps/camera/test/unit/controllers/recording-timer_test.js (hunk 3 above)
Comment (FunkTron) [inline apps/camera/test/unit/controllers/recording-timer_test.js]: Okay, sounds good

**Comment by try-server-hook** [comment]
[FunkTron FunkTron](https://github.com/FunkTron) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=51e6471be328)

**Comment by try-server-hook** [comment]
[FunkTron FunkTron](https://github.com/FunkTron) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=be3a97d137fd)

**Comment by try-server-hook** [comment]
[FunkTron FunkTron](https://github.com/FunkTron) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=3985c20124c7)

**Comment by try-server-hook** [comment]
[FunkTron FunkTron](https://github.com/FunkTron) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=7a5e9d0dcfb3)

**Comment by try-server-hook** [comment]
[FunkTron FunkTron](https://github.com/FunkTron) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=5f5b53bfa0d6)

File: apps/camera/style/controls.css [hunk 4]
```diff
@@ -43,7 +43,7 @@
 .controls-left {
   justify-content: flex-end;
   opacity: 1;
-  transition: opacity 200ms ease-in-out;
+  transition: opacity 200ms ease-in-out, visibility 0ms ease-in-out;
```
Comment (justindarc) [inline apps/camera/style/controls.css]: nit: for the 0ms `visibility` transition, its unnecessary to specify an easing function

File: apps/camera/style/recording-timer.css [hunk 5]
```diff
@@ -19,7 +20,7 @@
   text-align: left;
   pointer-events: none;
   opacity: 0;
-  transition: opacity 0.2s ease-in-out;
+  transition: opacity 0.2s ease-in-out, visibility 0s ease-in-out 0.2s;
```
Comment (justindarc) [inline apps/camera/style/recording-timer.css]: nit: same here -- I don't think its necessary to specify an easing function for `visibility`

**Comment by try-server-hook** [comment]
[FunkTron FunkTron](https://github.com/FunkTron) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=5c3b0bc503a5)

**Pull request merged**